azext_iot/central/models/templatepreview.py

```diff
@@ -77,18 +77,18 @@ def _extract_interfaces(self, template: dict) -> dict:
 
             interfaces = []
             dcm = template.get("capabilityModel", {})
-
+            
             if dcm.get("contents"):
                 interfaces.append(self._extract_root_interface_contents(dcm))
 
             if dcm.get("implements"):
                 interfaces.extend(dcm.get("implements"))
 
             return {
-                interface["@id"]: self._extract_schemas(interface)
+                (interface["schema"]["@id"] if interface.get("@type") else interface["@id"]): self._extract_schemas(interface)
```

Comment:
Can you move the code to determine and get the interface id into a separate helper function? It will help with readability.